lib/Dialect/Torch/Transforms/MaximizeValueSemantics.cpp

```diff
@@ -244,6 +288,111 @@ class AbstractlyInterpretCopyToNonValueTensorOpUsersWithinABlock
     rewriteSlice(*interpretedOps, rewriter);
     return success();
   }
+
+private:
+  static LogicalResult
+  calculateAliasInfo(Value tensor,
+                     DenseMap<Value, AliasSliceInfo> &aliasSliceInfo) {
+    Operation *op = tensor.getDefiningOp();
+
+    // If op is not a view-like op, then the tensor is an alias of itself.
+    if (!isViewLikeOp(op)) {
+      aliasSliceInfo[tensor] = AliasSliceInfo{tensor, -1, 0, -1, false};
+      return success();
+    }
```

Comment:
Once the recursion is removed, this should not be needed. I would turn it into an `assert`